src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts

```diff
@@ -1066,4 +1068,32 @@ function buildDotNetListItem(item: ListItem): DotNetListItem | null {
         children: children,
         actionSections: item.actionsSections
     } as DotNetListItem;
+}
+
+function checkConnectivity(viewId) {
+    let connectError = new Error('Cannot connect to ArcGIS Services!');
+    let message = '<div><h1>Cannot connect to ArcGIS Services.</h1><h2>Please check your internet connection.</h2></div>';
+    let mapContainer = document.getElementById(`map-container-${viewId}`)!; 
+    try {
+        if (!navigator.onLine) {
+            mapContainer.innerHTML = message;
+            throw new Error("Browser is offline! GeoBlazor requires an internet connection to work with ArcGIS.");
+        }
+        fetch('https://www.arcgis.com/sharing/rest')
```

Comment:
I'm not sure we should do this check. this guarantees that we are "online" however, It wouldn't be hard for someone to use the API behind their firewall (with no outside access) if we _don't_ check this. 

We are already bundling the ArcGIS JS API as needed, so I think the only external resource is this: `<link href="https://js.arcgis.com/4.24/esri/themes/light/main.css" rel="stylesheet"/>`.

If that's true, it's super easy for someone to "host" this themselves and not need "internet" access, only need access to their local network.

Does `navigator.online` correctly respond if you disable the internet in chrome dev tools?

Reply:
`navigator.online` was not reliable in my testing, which is why I implemented this.

Reply:
A few things that we discussed today that we should probably do in this PR:
- Host/deliver *all* of the JS assets (CSS, wasm, etc.) files. [Docs](https://developers.arcgis.com/javascript/latest/es-modules/#working-with-assets)
- Copy asset files on build
- Set the `config.assetsPath` property correctly.
- Optional: Expose other `config` properties that would let you host this on an offline/secured network: `assetsPath` and `fontsUrl` at a minimum, although we should probably expose all of these [config](https://developers.arcgis.com/javascript/latest/api-reference/esri-config.html#assetsPath) properties eventually (if we don't know, let's put them in a gissue for later).

I *think* that if we do all of this, a GeoBlazor app will run "disconnected" meaning that you don't have to have a basemap, and you can add graphics locally and it will work, draw the map etc. So you could run this "offline" although that's not really the purpose, it's really so you can run in an "isolated" network.

Reply:
Oh, and this means that we shouldn't need to check to see if `arcgis.com` is accessible from the code. If anything, we would check that the `assetsPath` is there, b/c that's what would allow the API itself to work. Any other online sources would just be for data. So we could still do this check, basically see if the `assetsPath` responds, and if it does we are "online" if not, display a message that assets can not be loaded.

@TimPurdum we'll still have to tell users to pick the correct CSS file (they have to pick the theme they want), not sure what that will look like now, but some local path?

Also, for deployment the user will have to ensure that the mime-types are setup on their server correctly ([docs](https://developers.arcgis.com/javascript/latest/install-and-set-up/#web-server-hosting-configuration))